File: app/src/main/java/net/programmierecke/radiodroid2/station/live/metadata/TrackMetadataSearcher.java
package net.programmierecke.radiodroid2.station.live.metadata;

import androidx.annotation.NonNull;

import net.programmierecke.radiodroid2.station.live.metadata.lastfm.LfmMetadataSearcher;

import okhttp3.OkHttpClient;

public class TrackMetadataSearcher {
    private LfmMetadataSearcher lfmMetadataSearcher;

    public TrackMetadataSearcher(OkHttpClient httpClient) {
        lfmMetadataSearcher = new LfmMetadataSearcher(httpClient);
    }


    public void fetchTrackMetadata(String artist, @NonNull String track, @NonNull TrackMetadataCallback trackMetadataCallback) {
        lfmMetadataSearcher.fetchTrackMetadata(artist, track, trackMetadataCallback);
    }
}

File: app/src/main/java/net/programmierecke/radiodroid2/station/live/metadata/TrackMetadataSearcher.kt
package net.programmierecke.radiodroid2.station.live.metadata

import net.programmierecke.radiodroid2.station.live.metadata.lastfm.LfmMetadataSearcher
import okhttp3.OkHttpClient

class TrackMetadataSearcher(httpClient: OkHttpClient) {
    private val metadataSearcher: LfmMetadataSearcher = LfmMetadataSearcher(httpClient)
    fun fetchTrackMetadata(artist: String, track: String, trackMetadataCallback: TrackMetadataCallback) =
            metadataSearcher.fetchTrackMetadata(artist, track, trackMetadataCallback)
}
